create_derivatives/routines.py

```diff
@@ -65,8 +67,104 @@ def get_as_uri(self, bag_filepath):
 
 
 class JP2Maker:
-    # TO DO: make JPG2000 derivatives
-    pass
+    """Creates JP2000 derivatives from TIFFs.
+
+    Creates JP2 directory in bag's data directory and JP2 derivatives.
+    TIFFs to be converted are targeted in the service directory. If the service directory is empty or
+    does not exist, target TIFFs at the root of the data directory.
+
+    Returns:
+        A tuple containing human-readable message along with list of bag identifiers.
+        Exceptions are raised for errors along the way.
+    """
+
+    def run(self):
+        bags_with_jp2s = []
+        for bag in Bag.objects.filter(process_status=Bag.PREPARED):
+            service_dir = Path(bag.bag_path, "data", "service")
```

Comment:
Would be curious to get @bonniegee's input on whether we need to check for a service directory or not. If we do I agree we need better tests, and we may want to break these lines into a `get_tiff_files_path` method so it can be tested in isolation.

Reply:
Yes, I think we'll need to check for a service directory. I think for special projects we'll still get preservation and mezzanine TIFF files, and will want to include both in our SIPs.

Reply:
Ok, if no one objects I'd like to create a separate issue for improving tests for this routine (that include testing with different directory locations of TIFFs) to move this PR forward.

Reply:
I support that approach!